app/src/main/java/org/kiwix/kiwixmobile/main/MainActivity.java

```diff
@@ -904,6 +918,115 @@ public boolean onOptionsItemSelected(MenuItem item) {
+      builder = new AlertDialog.Builder(this, R.style.AppTheme_Dialog_Night);
+    } else {
+      builder = new AlertDialog.Builder(this);
+    }
+
+    builder.setMessage(this.getString(R.string.delete_notes_confirmation_msg))
+        .setNeutralButton(this.getString(R.string.dismiss), new DialogInterface.OnClickListener() {
+          @Override
+          public void onClick(DialogInterface dialog, int which) {
+            // Do nothing
+          }
+        })
+        .setPositiveButton(this.getString(R.string.delete), new DialogInterface.OnClickListener() {
+          @Override
+          public void onClick(DialogInterface dialog, int which) {
+            clearAllNotes();
+          }
+        })
+        .show();
+  }
+
+  /** Method to delete all user notes */
+  private void clearAllNotes() {
+
+    boolean result = true; // Result of all delete() calls is &&-ed to this variable
+
+    if(AddNoteDialog.isExternalStorageWritable()) {
+      if(ContextCompat.checkSelfPermission(this, Manifest.permission.WRITE_EXTERNAL_STORAGE) != PackageManager.PERMISSION_GRANTED) {
+        Log.d("MainActivity", "WRITE_EXTERNAL_STORAGE permission not granted");
+        Toast.makeText(this, this.getString(R.string.ext_storage_permission_not_granted), Toast.LENGTH_LONG);
+        return;
+      }
+
+      File notesDirectory = new File(Environment.getExternalStorageDirectory() + "/Kiwix/Notes/");
+      File[] filesInNotesDirectory = notesDirectory.listFiles();
+
+      if(filesInNotesDirectory == null) { // Notes folder doesn't exist
+        Toast.makeText(this, this.getString(R.string.notes_deletion_none_found), Toast.LENGTH_LONG).show();
+        return;
+      }
+
+      for(File wikiFileDirectory : filesInNotesDirectory) {
+        if(wikiFileDirectory.isDirectory()) {
+          File[] filesInWikiDirectory = wikiFileDirectory.listFiles();
+
+          for(File noteFile : filesInWikiDirectory) {
+            if(noteFile.isFile()) {
+              result = result && noteFile.delete();
+            }
+          }
+        }
+
+        result = result && wikiFileDirectory.delete(); // Wiki specific notes directory deleted
+      }
+
+      result = result && notesDirectory.delete(); // "{External Storage}/Kiwix/Notes" directory deleted
+    }
+
+    if(result) {
+      Toast.makeText(this, this.getString(R.string.notes_deletion_successful), Toast.LENGTH_SHORT).show();
+    } else {
+      Toast.makeText(this, this.getString(R.string.notes_deletion_unsuccessful), Toast.LENGTH_SHORT).show();
+    }
+  }
+
+  /** Creates the full screen AddNoteDialog, which is a DialogFragment */
+  private void showAddNoteDialog() {
+    FragmentTransaction fragmentTransaction = getSupportFragmentManager().beginTransaction();
+    Fragment prev = getSupportFragmentManager().findFragmentByTag("AddNoteDialog");
+    if(prev != null) {
+      fragmentTransaction.remove(prev); // To prevent multiple instances of the DialogFragment
```

Comment:
I would prefer that you just wouldn't add a new fragment if one already exists instead of removing one because the user clicked the button again

Reply:
Refactored [here](https://github.com/kiwix/kiwix-android/pull/1198/commits/873eb2bce86c76b11962367701cb66bdf66b9f5e#diff-8f999695c436a978371f898c45907cdaR996)

Reply:
I could be wrong but I believe if we can find the dialogFragment by tag then it is already showing, what I wanted to see here was
`FragmentManager fragmentManager = getSupportFragmentManager();`
`FragmentTransaction fragmentTransaction = fragmentManager.beginTransaction();`
`if(fragmentManager.findFragmentByTag("AddNoteDialog") == null) {`
`    fragmentTransaction.addToBackStack(null); `
`    new AddNoteDialog().show(fragmentTransaction, "AddNoteDialog");`
`  }`
Also now that I look at it again, that magic string "AddNoteDialog" is a recipe for disaster, create a constant on AddNoteDialog called TAG and please use it here

Reply:
I checked and found that `findFragmentByTag()` looks for the fragment in the manager's activity, and if not found there, checks the back-stack. So there isn't any assurance that the `DialogFragment`, if found, is currently being displayed.

If the app functions as intended, then the AddNoteDialog will always exist at the top of the back-stack (since the parent activity cannot be accessed to create any other fragments and the dialog doesn't create any fragments of its own except the ConfirmationAlertDialogFragment). But if by some freak mistake/user-interaction it does end up below the top of the back-stack, `FragmentTransaction.show()` will ensure that it is brought to the top and displayed.

Reply:
Hmmm does it even need to be on the backstack? If we delete the adding it to the back stack then that is one more line gone and a branch of execution eliminated

Reply:
I'm showing a ConfirmationAlertDialogFragment on top of it, so I'll have to check this

Also shifted "AddNoteDialog" to a constant in the class [here](https://github.com/kiwix/kiwix-android/pull/1198/commits/7b8bd2b406b4b842166cf0f3937294fa97d35e94#diff-36e13fa4a9132616aee81e1f88bfa0ffR61)

Reply:
[Done](https://github.com/kiwix/kiwix-android/pull/1198/commits/62c46556229ae0bd3c4e27667545aed956ab0d68)
Didn't face any issues on not adding it to the back stack (orientation change, switching to night mode, shifting to another app and coming back, re-installation)